web/src/components/navbar/shared-submenu-components.tsx

```diff
@@ -0,0 +1,232 @@
+import { ButtonIcon } from "@/components/ButtonIcon";
+import { NavMenuItem } from "@/components/navbar/NavMenuItem";
+import { AuthContext } from "@/contexts/authContext";
+import { NeedsAccountContext } from "@/contexts/needsAccountContext";
+import { triggerSignIn } from "@/lib/auth/sessionHelper";
+import { onSelfRegister, onSignOut } from "@/lib/auth/signinHelper";
+import { useConfig } from "@/lib/data-hooks/useConfig";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { getBusinessIconColor } from "@/lib/domain-logic/getBusinessIconColor";
+import { getNavBarBusinessTitle } from "@/lib/domain-logic/getNavBarBusinessTitle";
+import { orderBusinessIdsByDateCreated } from "@/lib/domain-logic/orderBusinessIdsByDateCreated";
+import { QUERIES, ROUTES, routeWithQuery } from "@/lib/domain-logic/routes";
+import { switchCurrentBusiness } from "@/lib/domain-logic/switchCurrentBusiness";
+import analytics from "@/lib/utils/analytics";
+import { openInNewTab } from "@/lib/utils/helpers";
+import { useRouter } from "next/router";
+import { ReactElement, useContext } from "react";
+
+export const LoginMenuItem = (): ReactElement => {
+  const { Config } = useConfig();
+  return NavMenuItem({
+    onClick: (): void => {
+      analytics.event.landing_page_navbar_log_in.click.go_to_myNJ_login();
+      triggerSignIn();
+    },
+    icon: <ButtonIcon svgFilename="login" sizePx="25px" />,
+    itemText: Config.navigationDefaults.logInButton,
+    key: "loginMenuItem",
+  });
+};
+
+export const LogoutMenuItem = (props: { handleClose: () => void }): ReactElement => {
```

Comment:
dust - so I thought about breaking this out into many files and it just seemed kinda like pointless file bloat given these are all pretty simple in terms of implementation and testing (expcept profile which is more complicated)